### ISSUE TITLE
Initialization is not complete

### ISSUE THREAD

**Issue by veonua** [chris7topher/SmartPhilips2200#4]
When I send the power on command, it seems the chip doesn't send all the initialization commands needed.

The Hot Water test on my LatteGo 3200 ended up with a tray full of cold water. It doesn't even go through the regular pipe but dumps directly into the tray.

The coffee test ended up with cold coffee.

**Comment by micki88** [comment]
I have the same machine (LatteGo 3200) and expiriencing some strange behavior. 
I have managed to map all the commands from the display (they are different to the 2200 model). But when I connect the esp to the machine and say press espresso, after some seconds it starts on its own and then just stops the process.
Maybe errors in the forward of the serail.read to serial.write?
I am new to arduino...

**Comment by veonua** [comment]
@micki88 most likely you mapped commands wrong. do you have your code anywhere on GitHub

**Comment by micki88** [comment]
@veonua Hi https://github.com/micki88/Philips-ep3200-testing This is just Christophers code that I modified fo ep 3200.

**Comment by veonua** [comment]
my start commands are

  byte cmd1[] = {0xd5, 0x55, 0x0a, 0x01, 0x03, 0x00, 0x0d, 0x00, 0x00, 0x00, 0x1d, 0x36};
  byte cmd2[] = {0xd5, 0x55, 0x01, 0x01, 0x03, 0x00, 0x0d, 0x00, 0x00, 0x00, 0x36, 0x03};

I needed both to init successfully

**Comment by micki88** [comment]
Thx. My problem is the start/stop button/command. It is beeing randomly activated. So when I press something like coffee on the display and wait, it just starts on its own. Or if I start making coffee the machine randomly stops like if the stop button has been pressed. And I hear the beep sound, as you would press the start/stop button.
I dont understand how It is beeing activated on its own...

**Comment by veonua** [comment]
probably you need to check your wires, the start/stop button command is just one bit. it might be a noise

**Comment by micki88** [comment]
Thx veonua. I checked all wires again.  Still have same issue. Could the noise be coming from the unused softserial TX pin?

**Comment by micki88** [comment]
[Solved]
Just to inform, I have solved the issue  by swapping the serial and software serial communication. It now works. Thx again for help.

**Comment by sendorm** [comment]
Hello @micki88 so your codes works for ep3200 series right? Do all the settings work? The coffee types etc? Thanks

**Comment by veonua** [comment]
it's just an emulation of a button press by button number. 
I got a hidden “americano” discovered on my 2200 just by using the last known index +1 code

**Comment by micki88** [comment]
Yup all works. Would be cool to find out the code for "steam" on the latteGo machines like mine. So we could make hot milk without the need to stop the capuccino or latteMachiato program

**Comment by sendorm** [comment]
Do you have pictures of your setup and can you provide items list? People seem to be having issues with what to use. Which pin to connect etc? Maybe asking too much here :) Most of my experience is with esphome devices. Asked at their discord too if this can be implemented or not. Having this features on homeassistant will be great.

**Comment by walthowd** [comment]
I'm a 3200 machine user as well, and would be happy to collaborate somewhere

**Comment by veonua** [comment]
@walthowd just create a fork or a new project

**Comment by flashmandv** [comment]
Guys, does anyone know what could be the reason for the display not working when powered on from the ESP?
My Phillips is EP3246. Commands work fine but I can't power on the display. If I power on the machine from its display button, then the display is updated OK when I send any commands from the ESP. 

I checked the transistor and it works fine. It cuts the ground to the display for 300ms (tried different values). I also tried to cut the display GND manually or before/after sending the power on command. Nothing works. Please help me power on the display from the ESP.
Thanks

**Comment by micki88** [comment]
@flashmandv probably wrong command or you maybe need two commands. Try command from my fork for 3200 series.

**Comment by flashmandv** [comment]
@micki88 the commands from your repo should be fine as the machine turns on just fine. It is just the display that refuses to turn on :/ Maybe for my machine the cutout to ground for 300ms is not working..but it is strange as this works for the other machines from 3200 series.

**Comment by micki88** [comment]
@flashmandv Ok, just to be sure, you did cut the other cables? I think I remember having two cables to ground on the display...

**Comment by micki88** [comment]
![Screenshot_20240101_131610_Samsung Internet.jpg](https://github.com/chris7topher/SmartPhilips2200/assets/77992953/666f9673-89c3-4689-88d8-5d8b8907fdd8)

Can you try this power on commands

**Comment by flashmandv** [comment]
> @flashmandv Ok, just to be sure, you did cut the other cables? I think I remember having two cables to ground on the display...

Actually I haven't cut all the wires to the display. I only cut the Tx and the GND (that is next to the 5V).
Do I need to cut the other unused cables? The 3th cable (that is again GND) needs to be cut completely? or wired to the transistor so he cuts it momentarily?

Thanks for the new commands. I will try them today

**Comment by flashmandv** [comment]
Thank you @micki88 It worked when I cut the 3th ground pin and connected it to the transistor emitter.
The power on command also worked (but my old one was also working..I'm not sure which is the best one but they both work).
Now it remained to clear some strange bugs (setting water level and strength level does not work properly..maybe because the machine update statis is not always updating)

<img width="491" alt="image" src="https://github.com/chris7topher/SmartPhilips2200/assets/2466808/e6b61ad4-dc45-43fc-ac1a-db0a70280d6f">
